test/hotspot/jtreg/compiler/loopopts/superword/TestMemorySegment.java

```diff
@@ -795,6 +826,13 @@ static Object[] testLongLoop_longIndex_intInvar_int(MemorySegment a, int invar)
     @IR(counts = {IRNode.LOAD_VECTOR_I, "= 0",
                   IRNode.ADD_VI,        "= 0",
                   IRNode.STORE_VECTOR,  "= 0"},
+        applyIfAnd = { "ShortRunningLongLoop", "false", "AlignVector", "false" },
+        applyIfPlatform = {"64-bit", "true"},
+        applyIfCPUFeatureOr = {"sse4.1", "true", "asimd", "true", "rvv", "true"})
+    @IR(counts = {IRNode.LOAD_VECTOR_I, "> 0",
+                  IRNode.ADD_VI,        "> 0",
+                  IRNode.STORE_VECTOR,  "> 0"},
+        applyIfAnd = { "ShortRunningLongLoop", "true", "AlignVector", "false" },
```

Comment:
These changes weren't perfect, now we are not covering all cases with the IR rules... I'll see if I can fix that with the patch for the aliasing runtime check.